activerecord/lib/active_record/connection_adapters/postgresql/schema_definitions.rb

```diff
@@ -42,7 +42,12 @@ module ColumnMethods
         # a record (as primary keys cannot be +nil+). This might be done via the
         # +SecureRandom.uuid+ method and a +before_save+ callback, for instance.
         def primary_key(name, type = :primary_key, **options)
-          options[:default] = options.fetch(:default, "gen_random_uuid()") if type == :uuid
+          if type == :uuid
+            options[:default] = options.fetch(:default, "gen_random_uuid()")
+          elsif options.delete(:auto_increment) == true && type == :integer
+            type = :serial
```

Comment:
Yes, that was my thought. It seems useful (and just more understandable) to keep serial and bigserial balanced, IMO.. even if we're not sure about exposing the option.